rapid7/metasploit-framework
# Beginnings of the Android Interval collection functionality

As per the MSF Payloads PR here: https://github.com/rapid7/metasploit-payloads/pull/20

This contains the MSF-side of the changes required to make the Android interval collection functionality work. Most of the detail of this can be found in the Payloads PR.

The interface allows for the collector(s) to be started, stopped, paused and resumed. There's a dump command that will take any data currently collected and render it to screen. The stop command stops the collector _and_ returns any data that has been captured.
## Sample Run

Help:

```
meterpreter > interval_collect 
Usage: interval_collect <parameters>

Specifies an action to perform on a collector type.


OPTIONS:

    -a <opt>  Action (required, one of: start, pause, resume, stop, dump)
    -c <opt>  Collector type (required, one of: wifi)
    -h        Help Banner
    -t <opt>  Collect poll timeout period in seconds (default: 30)
```

Creation of an interval collector that periodically pulls wifi detail down:

```
meterpreter > interval_collect -t 15 -c wifi -a start
[+] Interval action completed successfully
meterpreter > interval_collect -c wifi -a stop

Captured wifi data
==================

Last Seen            BSSID              SSID           Level
---------            -----              ----           -----
2015-08-18 17:27:05  XX:XX:XX:XX:XX:XX  TheColonial    -40
2015-08-18 17:27:12  XX:XX:XX:XX:XX:XX  ASLR           -51
2015-08-18 17:27:12  XX:XX:XX:XX:XX:XX  HeapSpray-x64  -45
2015-08-18 17:27:12  XX:XX:XX:XX:XX:XX  HeapSpray      -37
```
## Verification

So far we only have the one collector, but there's more to come.
1. Pull and build the Payload PR from: https://github.com/rapid7/metasploit-payloads/pull/20
2. Get an android meterpreter session.
3. Kick off the wifi interval collector using `interval_collect -c wifi -a start`  (the use of the `-t` for timeout is optional.. I'd set it to something like `10` if you dont' want to wait long).
4. Wait for the `n` seconds and then use `interval_collect -c wifi -a dump` to see the dump functioanlity work.
5. Wait again for the `n` seconds and use `interval_collect -c wifi -a stop`. This does the same as dump, but also stops the collector.
6. Fire up the collector again and make sure that `interval_collect -c wifi -a pause` and `interval_collect -c wifi -a resume` work as expect.
